src/polymer/polymer-element.ts

```diff
@@ -262,9 +262,9 @@ function resolveElement(
     scannedElement: ScannedPolymerElement, document: Document): PolymerElement {
   const element = new PolymerElement();
   element.privacy = scannedElement.privacy;
-  applySuperClass(element, scannedElement, document);
-  applyMixins(element, scannedElement, document);
   applySelf(element, scannedElement, document);
+  applyMixins(element, scannedElement, document);
+  applySuperClass(element, scannedElement, document);
```

Comment:
I see, with how _inheritVals works below. I think that it's a little easier to follow by sticking to modeling the prototype chain and building a class from top-down. This also leaves room to cache prototypes. The only performance hit is due to overwriting overloaded members, but I think that's going to be a very small percentage of members.